docs/nddata/nddata.rst

```diff
@@ -148,67 +255,108 @@ example, a FITS header can be used as the metadata::
     10
 
-Conflicting parameters
-----------------------
-
-It is possible to initialize `~astropy.nddata.NDData` with an explicit
-parameter in the initialization call and an implicit one in the ``data``
-parameter. If such a combination is detected a warning is
-issued and the explicit parameter is kept (without any attempt of conversion).
+.. note::
+    In some cases setting ``copy=True`` will copy the ``data`` twice. Known
+    cases are if the ``data`` is a `list` or `tuple`.
 
-For example with quantities::
+Converting NDData to other classes
+----------------------------------
 
-    >>> import astropy.units as u
-    >>> quantity = np.array([1, 2, 3, 4]) * u.m
-    >>> ndd = NDData(quantity, unit="cm")  # Explicit unit is cm and implicit is m, keep explicit.
-    INFO: Overwriting Quantity's current unit with specified unit [astropy.nddata.nddata]
-    >>> ndd.unit
-    Unit("cm")
+There is limited to support to convert a `~astropy.nddata.NDData` instance to
+other classes. In the process some properties might be lost.
 
-But this can affect the mask if the ``data`` parameter is a `~numpy.ma.MaskedArray`
-or even all attributes if the ``data`` is another `~astropy.nddata.NDData`
-instance.
+    >>> data = np.array([1, 2, 3, 4])
+    >>> mask = np.array([True, False, False, True])
+    >>> unit = 'm'
+    >>> ndd = NDData(data, mask=mask, unit=unit)
 
-Converting to Numpy arrays
---------------------------
+`numpy.ndarray`
+^^^^^^^^^^^^^^^
 
-Data should be accessed through the ``data`` attribute::
+Converting the ``data`` to an array::
 
     >>> array = np.asarray(ndd.data)
+    >>> array
+    array([1, 2, 3, 4])
+
+Though using ``np.asarray`` is not required in most cases it will ensure that
+the result is always a `numpy.ndarray`
+
+`numpy.ma.MaskedArray`
+^^^^^^^^^^^^^^^^^^^^^^
+
+Converting the ``data``  and ``mask`` to a MaskedArray::
+
+
+    >>> masked_array = np.ma.array(ndd.data, mask=ndd.mask)
+    >>> masked_array
+    masked_array(data = [-- 2 3 --],
+                 mask = [ True False False  True],
+           fill_value = 999999)
+
+`~astropy.units.Quantity`
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Converting the ``data``  and ``unit`` to a Quantity::
+
+    >>> quantity = u.Quantity(ndd.data, unit=ndd.unit)
+    >>> quantity
+    <Quantity [ 1., 2., 3., 4.] m>
+
+masked Quantity
+^^^^^^^^^^^^^^^
+
+Converting the ``data``, ``mask``  and ``unit`` to a masked Quantity::
 
-Though using ``np.asarray`` is not required it will ensure that an additional
-copy of the data is not made if the data is a numpy array.
+    >>> ma_quantity = np.ma.array(u.Quantity(ndd.data, unit=ndd.unit), mask=ndd.mask)  # doctest: +SKIP
+    >>> ma_quantity  # doctest: +SKIP
+    masked_Quantity(data = [-- 2.0 3.0 --] m,
+                    mask = [ True False False  True],
+              fill_value = 1e+20)
 
-Note that if the data is masked you must explicitly construct a numpy masked
-array like this::
+.. warning::
```

Comment:
I would change this to a note or incorporate it into the sentence above the example.